tidy3d/plugins/smatrix/ports/coaxial_lumped.py

```diff
@@ -0,0 +1,381 @@
+
+    def compute_current(self, sim_data: SimulationData) -> FreqDataArray:
+        """Helper to compute current flowing through the port."""
+        # Loops around inner conductive circle conductor
+        field_data = sim_data[f"{self.name}_current"]
+
+        # Helper for generating x,y vertices around a circle in the local coordinate frame
+        def generate_circle_coordinates(radius, num_points):
+            angles = np.linspace(0, 2 * np.pi, num_points, endpoint=True)
+            xt = radius * np.cos(angles)
+            yt = radius * np.sin(angles)
+            return (xt, yt)
+
+        # Get transverse axes
+        trans_axes = self.remaining_axes
+        coord1 = "xyz"[trans_axes[0]]
+        coord2 = "xyz"[trans_axes[1]]
+        coord3 = "xyz"[self.normal_axis]
+
+        # Just need a rough estimate for the number of cells
+        field_coords = field_data.field_components["H" + coord1].coords
+
+        # Estimate number of points needed along path integral
+        num_coords_1 = len(field_coords[coord1].values)
+        num_coords_2 = len(field_coords[coord2].values)
+        num_coords = max(num_coords_1, num_coords_2)
+        num_path_coords = round(np.pi * num_coords / 4) * 4 + 1
+
+        # These x,y coordinates are relate to the local coordinate frame
+        xt, yt = generate_circle_coordinates(
+            (self.outer_diameter + self.inner_diameter) / 4, num_path_coords
+        )
+        xt += self.center[trans_axes[0]]
+        yt += self.center[trans_axes[1]]
+
+        circle_vertices = np.column_stack((xt, yt))
+        # Close the contour exactly
+        circle_vertices[-1, :] = circle_vertices[0, :]
+
+        # Get the coordinates normal to port and select positions just on either side of the port
+        normal_coords = field_coords[coord3].values
+        upper_bound = np.searchsorted(normal_coords, self.center[self.normal_axis])
+        lower_bound = upper_bound - 1
+        # We need to choose which side of the port to place the path integral,
+        # which depends on which way the inner conductor is extending
+        if self.direction == "+":
+            path_pos = normal_coords[upper_bound]
+        else:
+            path_pos = normal_coords[lower_bound]
+
+        # Setup the path integral and integrate the H field
+        path_integral = CustomCurrentIntegral2D(
+            axis=self.normal_axis, position=path_pos, vertices=circle_vertices
+        )
+        current = path_integral.compute_current(field_data)
+
+        # We need the current flowing transverse through the port, which is the opposite of
+        # the current flowing in the core conductor in the positive direction.
+        if self.direction == "+":
+            current *= -1.0
+
+        return current
+
+    @cached_property
+    def remaining_axes(self) -> Tuple[Axis, Axis]:
+        """Get in-plane axes, ordered to maintain a right-handed coordinate system."""
+        axes: List[Axis] = [0, 1, 2]
```

Comment:
I might be wrong, but I think for python 12 support, we need to use `list[Axis]` now (not typing.List), although this might not apply to method annotations.

Reply:
Should I also extend that change to things like `tuple`?

Reply:
oh right, yea `tuple[Axis, Axis]`